code/modules/games/cards.dm

```diff
@@ -0,0 +1,443 @@
+				cards += P
+			qdel(H)
+			to_chat(user,"<span class='notice'>You place your cards on the bottom of [src]</span>.")
+			return
+		else
+			to_chat(user,"<span class='warning'>You can't mix cards from other deck!</span>")
+			return
+
+	..()
+
+/obj/item/weapon/deck/attack_hand(mob/user as mob)
+	draw_card(user)
+
+/obj/item/weapon/deck/verb_pickup() // Speshul Snowflek Override for card so pickup work as intended.
+	if(!(usr)) //BS12 EDIT
+		return
+	if(!usr.canmove || usr.stat || usr.restrained() || !Adjacent(usr))
+		return
+	if((!istype(usr, /mob/living/carbon)) || (istype(usr, /mob/living/carbon/brain)))//Is humanoid, and is not a brain
+		to_chat(usr, "<span class='warning'>You can't pick things up!</span>")
+		return
+	if(usr.stat || usr.restrained() )//Is not asleep/dead and is not restrained
+		to_chat(usr, "<span class='warning'>You can't pick things up!</span>")
+		return
+	if(anchored) //Object isn't anchored
+		to_chat(usr, "<span class='warning'>You can't pick that up!</span>")
+		return
+	if(!usr.hand && usr.r_hand) //Right hand is not full
+		to_chat(usr, "<span class='warning'>Your right hand is full.</span>")
+		return
+	if(usr.hand && usr.l_hand) //Left hand is not full
+		to_chat(usr, "<span class='warning'>Your left hand is full.</span>")
+		return
+	if(!istype(src.loc, /turf)) //Object is on a turf
+		to_chat(usr, "<span class='warning'>You can't pick that up!</span>")
+		return
+	//All checks are done, time to pick it up!
+	usr.put_in_hands()
+	return
+
+/obj/item/weapon/deck/verb/draw_card(mob/user as mob)
+
+	set category = "Object"
+	set name = "Draw"
+	set desc = "Draw a card from a deck."
+	set src in view(1)
+
+	var/mob/living/carbon/human/M = user
+
+	if(user.stat || !Adjacent(user))
+		return
+
+	if(!cards.len)
+		to_chat(user,"<span class='notice'>There are no cards in the deck.</span>")
+		return
+
+	var/obj/item/weapon/hand/H = M.is_in_hands(/obj/item/weapon/hand)
+	if(H && !(H.parentdeck == src))
+		to_chat(user,"<span class='warning'>You can't mix cards from other deck!</span>")
+		return
+
+	if(!H)
+		H = new(get_turf(src))
+		user.put_in_hands(H)
+
+	if(!H || !user) return
+
+	var/datum/playingcard/P = cards[1]
+	H.cards += P
+	cards -= P
+	H.parentdeck = src
+	H.update_icon()
+	user.visible_message("<span class='notice'>[user] draws a card.</span>")
+	to_chat(user,"<span class='notice'>It's the [P].</span>")
+
+/obj/item/weapon/deck/verb/deal_card()
+
+	set category = "Object"
+	set name = "Deal"
+	set desc = "Deal a card from a deck."
+	set src in view(1)
+
+	if(usr.stat || !Adjacent(usr))
+		return
+
+	if(!cards.len)
+		to_chat(usr,"<span class='notice'>There are no cards in the deck.</span>")
+		return
+
+	var/list/players = list()
+	for(var/mob/living/player in viewers(3))
+		if(!player.stat)
+			players += player
+	//players -= usr
+
+	var/mob/living/M = input("Who do you wish to deal a card?") as null|anything in players
+	if(!usr || !src || !M) return
+
+	deal_at(usr, M, 1)
```

Comment:
Needs some kind of movement check so you can't move while the popup is open and deal people cards from afar